.github/workflows/docker.yml

```diff
@@ -0,0 +1,62 @@
+---
+name: Docker
+
+on:
+  push:
+    branches:
+      - eodatasets3
+    paths:
+      - "**"
+
+  release:
+    types: [created, edited, published]
+
+env:
+  IMAGE_NAME: opendatacube/eo-datasets
+
+jobs:
+  docker:
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/eodatasets3' || github.event_name == 'release'
+
+    steps:
+      - name: Checkout git
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Get unstable git tag
+        run: >
+          echo "UNSTABLE_TAG=$(git describe --tags | sed 's/eodatasets3-//')" >> $GITHUB_ENV
+
+      - name: Log the unstable tag
+        run: echo $UNSTABLE_TAG
+
+      - name: Build and Push unstable + latest Docker image tag
+        if: github.event_name != 'release'
+        uses: whoan/docker-build-with-cache-action@v4
```

Comment:
This is probably not recommended any more.

Check this out. https://github.com/digitalearthafrica/deafrica-tools/blob/main/.github/workflows/build-test-docker.yml

Up to you which way you choose to go!

Reply:
Ooh, thanks for pointing it out. I was just cargo-culting without reading the docs myself.

I ended up removing caching, as this build step is for publishing images, not for running tests, and with the gdal base image it seems to only take a minute to build anyway.

Also switched it to use the `docker/metadata-action` to extract the version number / tags, as it looks like it has nicer defaults. I'd like to move this repo to use more typical branch+tag naming schemes anyway, so may as well do that now.